### PR TITLE
feat(user): Init user module

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add blank slate for user module

### CODE REVIEW

File: packages/types/src/user/service.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { CreateUserDTO, UpdateUserDTO } from "./mutations"
+
+export interface IUserModuleService extends IModuleService {
+  retrieveUser(
+    id: string,
+    config?: FindConfig<UserDTO>,
+    sharedContext?: Context
+  ): Promise<UserDTO>
+
+  listUsers(
```
Comment (carlos-r-l-rodrigues) [inline packages/types/src/user/service.ts]: These are methods to retrieve and list from the main entity, I believe it is fine to keep them without the suffix (User).
`retrieve, list, listAndCount`

Comment (adrien2p) [reply packages/types/src/user/service.ts]: Yes indeed, like the auto generated methods will be. I suggest you to glance over my pr description for the abstraction, i added all those conventions as well 👍

Comment (olivermrbl) [reply packages/types/src/user/service.ts]: We should probably await your PR before merging this one?

File: packages/types/src/user/mutations.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export type CreateUserDTO = { id?: string }
+export type UpdateUserDTO = { id: string }
```
Comment (adrien2p) [inline packages/types/src/user/mutations.ts]: suggestioin: Dont forget to move to interfaces later 👍

File: packages/user/src/migrations/Migration20240201081925.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Migration } from '@mikro-orm/migrations';
+
+export class Migration20240201081925 extends Migration {
+
+  async up(): Promise<void> {
+    this.addSql('create table "user" ("id" text not null, constraint "user_pkey" primary key ("id"));');
```
Comment (adrien2p) [inline packages/user/src/migrations/Migration20240201081925.ts:6]: comment: will probably change a lot later, but not to forget the if exists and dissociate the table creation and the columns

**Review by adrien2p — APPROVED** [review]
LGTM 🔥

File: packages/user/src/migrations/Migration20240201081925.ts (hunk 3 above)
Comment (pKorsholm) [reply packages/user/src/migrations/Migration20240201081925.ts:6]: yes, will update the table definitions in a subsequent pr (including `if exists`)

File: packages/user/integration-tests/__fixtures__/user.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { SqlEntityManager } from "@mikro-orm/postgresql"
```
Comment (srindom) [inline packages/user/integration-tests/__fixtures__/user.ts:1]: thought(non-blocking): this approach of having one big seeder feels a bit off - think creating the data for the test within the test case is easier to wrap you your head around + you can use the service methods directly.

**Review by srindom — APPROVED** [review]
LGTM

File: packages/user/src/migrations/Migration20240201081925.ts (hunk 3 above)
Comment (srindom) [reply packages/user/src/migrations/Migration20240201081925.ts:6]: maybe worth leaving the migration out entirely for now

File: packages/user/integration-tests/__fixtures__/user.ts (hunk 4 above)
Comment (pKorsholm) [reply packages/user/integration-tests/__fixtures__/user.ts:1]: Will update when introducing models today 😄

**Pull request merged**